Sources/ArcGISToolkit/Components/Offline/PreplannedMapModel.swift

```diff
@@ -172,7 +186,10 @@ class PreplannedMapModel: ObservableObject, Identifiable {
             let result = await job.result
             guard let self else { return }
             self.updateDownloadStatus(for: result)
-            self.mobileMapPackage = try? result.map { $0.mobileMapPackage }.get()
+            if status.isDownloaded {
+                self.mobileMapPackage = try? result.get().mobileMapPackage
+                self.directorySize = FileManager.default.sizeOfDirectory(at: mmpkDirectoryURL)
```

Comment:
The size is still being calculated on the main thread, but at least it isn't being done in a view update anymore.

Reply:
I'm not entirely sure how to "dispatch" a long-running synchronous operation onto a background thread in Swift Concurrency. Do you have a specific suggestion in mind?

Reply:
No, I don't. Remember that not every comment is actionable. This was just thinking out load.